lib/ansible/modules/remote_management/oneview/oneview_datacenter.py

```diff
@@ -0,0 +1,178 @@
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: oneview_datacenter
+short_description: Manage OneView Data Center resources
+description:
+    - "Provides an interface to manage Data Center resources. Can add, update, and remove."
+version_added: "2.5"
+requirements:
+    - "python >= 2.7.9"
+    - "hpOneView >= 3.1.0"
+author:
+    - Alex Monteiro (@aalexmonteiro)
+    - Madhav Bharadwaj (@madhav-bharadwaj)
+    - Priyanka Sood (@soodpr)
+    - Ricardo Galeno (@ricardogpsf)
+options:
+    state:
+        description:
+            - Indicates the desired state for the Data Center resource.
+              C(present) will ensure data properties are compliant with OneView.
+              C(absent) will remove the resource from OneView, if it exists.
+        choices: ['present', 'absent']
+        required: true
+    data:
+        description:
+            - List with Data Center properties and its associated states.
+        required: true
+
+extends_documentation_fragment:
+    - oneview
+    - oneview.validateetag
+'''
+
+EXAMPLES = '''
+- name: Add a Data Center
+  oneview_datacenter:
+    hostname: 172.16.101.48
+    username: administrator
+    password: my_password
+    api_version: 500
+    state: present
+    data:
+        contents:
+            # You can choose either resourceName or resourceUri to inform the Rack
+            - resourceName: '{{ datacenter_content_rack_name }}' # option 1
+              resourceUri: ''                                    # option 2
+              x: 1000
+              y: 1000
+        depth: 6000
+        name: "MyDatacenter"
+        width: 5000
+  no_log: true
+  delegate_to: localhost
+
+- name: Update the Data Center with specified properties (no racks)
+  oneview_datacenter:
+    hostname: 172.16.101.48
+    username: administrator
+    password: my_password
+    api_version: 500
+    state: present
+    data:
+        contents: ~
+        coolingCapacity: '5'
+        coolingMultiplier: '1.5'
+        costPerKilowattHour: '0.10'
+        currency: USD
+        defaultPowerLineVoltage: '220'
+        deratingPercentage: '20.0'
+        depth: 5000
+        deratingType: NaJp
+        name: "MyDatacenter"
+        width: 4000
+  no_log: true
+  delegate_to: localhost
+
+- name: Rename the Data Center
+  oneview_datacenter:
+    hostname: 172.16.101.48
+    username: administrator
+    password: my_password
+    api_version: 500
+    state: present
+    data:
+        name: "MyDatacenter"
+        newName: "My Datacenter"
+  no_log: true
+  delegate_to: localhost
+
+- name: Remove the Data Center
+  oneview_datacenter:
+    hostname: 172.16.101.48
+    username: administrator
+    password: my_password
+    api_version: 500
+    state: absent
+    data:
+        name: 'My Datacenter'
+  no_log: true
+  delegate_to: localhost
+'''
+
+RETURN = '''
+datacenter:
```

Comment:
Since this variable is in singular I'm assuming this module can't get all datacenters, am I right?

Reply:
I'm guessing this is because the module will manage one datacenter at a time, and return that datacenter's data. For getting multiple datacenters, likely the DatacenterFactsModule will be required.